applications/tari_watcher/src/forker.rs

```diff
@@ -0,0 +1,190 @@
+                web_ui_address.unwrap(),
+            )
+            .await?;
+
+        // TODO: stdout logs
+        // let process_dir = self.base_dir.join("processes").join("TariValidatorNode");
+        // let stdout_log_path = process_dir.join("stdout.log");
+        // let stderr_log_path = process_dir.join("stderr.log");
+        command
+            .kill_on_drop(true)
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .stdin(Stdio::null());
+
+        let child = command.spawn()?;
+
+        Ok(child)
+    }
+
+    async fn get_command(
+        &self,
+        target_binary: PathBuf,
+        network: String,
+        base_node_grpc_address: String,
+        json_rpc_public_address: String,
+        web_ui_address: String,
+    ) -> anyhow::Result<Command> {
+        log::debug!("Creating validator command from base directory: {:?}", self.base_dir);
+
+        // create directory for the validator process
+        let process_dir = self.base_dir.join("processes").join("TariValidatorNode");
+        fs::create_dir_all(&process_dir).await?;
+
+        log::debug!("Creating validator process to run from: {:?}", process_dir);
+
+        let json_rpc_address = json_rpc_public_address.clone();
+        let mut command = Command::new(target_binary);
+        let empty: Vec<(&str, &str)> = Vec::new();
+        command
+            .envs(empty)
+            .arg("-b")
+            .arg(process_dir)
+            .arg("--network")
+            .arg(network)
+            .arg(format!("--json-rpc-public-address={json_rpc_public_address}"))
+            .arg(format!(
+                "-pvalidator_node.base_node_grpc_address={base_node_grpc_address}"
+            ))
+            .arg(format!("-pvalidator_node.json_rpc_listener_address={json_rpc_address}"))
+            .arg(format!("-pvalidator_node.http_ui_listener_address={web_ui_address}"))
+            .arg("-pvalidator_node.base_layer_scanning_interval=1");
+        Ok(command)
+    }
+}
+
+#[allow(dead_code)]
+#[derive(Clone)]
+struct Instance {
+    app: InstanceType,
+    config: ExecutableConfig,
+    listen_ip: Option<IpAddr>,
+    port: PortAllocator,
+}
+
+impl Instance {
+    pub fn new(app: InstanceType, config: ExecutableConfig) -> Self {
+        Self {
+            app,
+            config,
+            listen_ip: None,
+            port: PortAllocator::new(),
+        }
+    }
+
+    pub async fn bootstrap(&mut self) -> io::Result<()> {
+        if self.listen_ip.is_some() {
+            log::warn!("Instance {} already bootstrapped, ignore", self.app.to_string());
+            return Ok(());
+        }
+
+        let listen = IpAddr::V4(Ipv4Addr::from([127, 0, 0, 1]));
+        self.listen_ip = Some(listen);
+
+        match self.app {
+            TARI_L2_VN => {
+                self.port.open_at(TARI_L2_VN, "jrpc").await?;
+                self.port.open_at(TARI_L2_VN, "web").await?;
+            },
+            MINO_L1_WALLET => {
```

Comment:
There is no need to monitor the minotari wallet process. 
Either, we should assume the minotari wallet is running with gRPC call to register a VN is available. The watcher simply needs to have the gRPC address to call. We can 100% do this if it's easier. 

Or optionally, we could run a command to send the registration [see here](https://github.com/tari-project/tari/blob/35573840c3b40f7e87b4021f8bed754be8e859cf/applications/minotari_console_wallet/src/cli.rs)

The watcher would not monitor the process but simply execute the command when we need to register.

Happy for either approach.

Reply:
> The watcher simply needs to have the gRPC address to call. We can 100% do this if it's easier.

Sounds like this is the easier approach for now and we can iterate later if needed